### ISSUE TITLE
Blazor webasm - EventCallBack with async handler show compiler error CS0407 or CS0123

### ISSUE THREAD

**Issue by LaughingJohn** [dotnet/roslyn#85403]
Hi,

I have a component with an EventCallBack which has a signature of:

`[Parameter] public EventCallback<MouseEventArgs> OnClick { get; set; }`

When I add an async event handler:

```
<ToolbarButton OnClick="@CopyCompanyAddress"
                          Label="Copy Company Address"
                          SvgFolder="SvgFolder.Buttons"
                          SvgName="copycompanyaddress" />
```

Where the handler looks like this:

`private async Task CopyCompanyAddress(MouseEventArgs args)`

I get an error message and squiggly telling me this:

`Error	CS0123	No overload for 'CopyCompanyAddress' matches delegate 'Action'`

If I change the handler to remove the the MouseEventArgs (which I don't actually need) like this:

`private async Task CopyCompanyAddress()`

I get the following error message:

`Error	CS0407	'Task FmFormItem.CopyCompanyAddress()' has the wrong return type`

I was under the impression it was OK to add a Task async handler for EventCallBack? Non-async void handlers don't show these errors, nor does using async void. 

I'm using VS2019 16.9 preview 1.  Everything appears to compile and run, but the error message is making me nervous. Am I doing something wrong or is the error just misleading?

Thanks.
